### PR TITLE
Importer now use FileExtensions enum, too, FileDialogs display extension

### PR DESCRIPTION
**Pull request by Siedlerchr**
Grouped Importer in FileExtension
Created test for FileExtensions
File Dialogs now display description and extension 

(internal change)
- [X] Tests created for changes
- [X] Screenshots added (for bigger UI changes)
- [X] Manually tested changed features in running JabRef

filefilter

### CODE REVIEW

File: src/main/java/net/sf/jabref/gui/FileExtensions.java [hunk 1]
```diff
@@ -12,36 +14,34 @@
...
+    REPEC("REPEC New Economic Papers (NEP)", "txt"),
+    RIS(Localization.lang("%0 file", "RIS"), "ris"),
+    SILVER_PLATTER(Localization.lang("%0 file", "SilverPlatter"), "dat", "txt"),
+    XMP(Localization.lang("XMP-annotated PDF"), "pdf"),
+
+    AUX(Localization.lang("%0 file", "AUX"), "aux"),
+    JSTYLE(Localization.lang("Style file"), "jstyle"),
+    LAYOUT(Localization.lang("Custom layout file"), "layout"),
+    TERMS(Localization.lang("Protected terms file"), "terms"),
+    TXT(Localization.lang("%0 file", Localization.lang("plain text")), "txt"),
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/FileExtensions.java]: There is already an old translation string for "plain text", right? If not, "Plain text" would be preferred.

Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/FileExtensions.java]: Or maybe even go for "Plain text file" now that I read the whole line of code. :-)

File: src/main/java/net/sf/jabref/gui/importer/ImportFormats.java [hunk 2]
```diff
@@ -26,6 +26,7 @@
 import javax.swing.Action;
 import javax.swing.JFileChooser;
 import javax.swing.JOptionPane;
+
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/importer/ImportFormats.java]: Checkstyle will complain (I guess).

File: src/main/java/net/sf/jabref/gui/FileExtensions.java [hunk 3]
```diff
@@ -12,36 +14,34 @@
 public enum FileExtensions {
```
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/FileExtensions.java]: Slightly unrelated, but didn't @tobiasdiez have some good argumentation that this type of file should be in singular? So `FileExtension`. Not in any way required for the PR though.

**Comment by oscargus** [comment]
Looks good to me! Some minor comments, but nothing crucial (and I guess Checkstyle will sort one of them out :-)).

File: src/main/java/net/sf/jabref/gui/FileExtensions.java (hunk 1 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/FileExtensions.java]: Ah I just noticed, we have already both cases of Plain text in the translations.

File: src/main/java/net/sf/jabref/gui/FileExtensions.java (hunk 3 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/FileExtensions.java]: I already thought about it when doing it first in  #1336 and would generally agree, but in this case one entry can have multiple extensions (for example EndNote) and I therefore thought the plural form would reflect it better

**Comment by Siedlerchr** [comment]
Anything else, otherwise I would merge it in. (Codacy probably failed because I rebased/squashed some things)

**Pull request merged**